zebrad/tests/acceptance.rs

```diff
@@ -2606,3 +2605,86 @@ async fn state_format_test(
     }
     Ok(())
 }
+
+/// Snapshot the `z_getsubtreesbyindex` method in a synchronized chain.
+#[tokio::test]
+#[ignore]
+async fn rpc_z_getsubtreesbyindex_sync_snapshots() -> Result<()> {
+    let _init_guard = zebra_test::init();
+
+    // We're only using cached Zebra state here, so this test type is the most similar
+    let test_type = TestType::UpdateZebraCachedStateWithRpc;
+    let network = Network::Mainnet;
+
+    let (mut zebrad, zebra_rpc_address) = if let Some(zebrad_and_address) = spawn_zebrad_for_rpc(
+        network,
+        "rpc_z_getsubtreesbyindex_sync_snapshots",
+        test_type,
+        true,
+    )? {
+        tracing::info!("running fully synced zebrad z_getsubtreesbyindex RPC test");
+
+        zebrad_and_address
+    } else {
+        // Skip the test, we don't have the required cached state
+        return Ok(());
+    };
+
```

Comment:
Since this test requires the subtree upgrade to be complete, we'll need to wait for it.

These functions are added in PR #7566, so you'll need to wait for it to merge, or copy them across to this PR.

This code gets the log message that says if an upgrade is running, it needs to be run after `zebrad` is launched:
https://github.com/ZcashFoundation/zebra/blob/be6a33ee6598f7cd86cbd1da7d961a606c88d07b/zebrad/tests/common/lightwalletd/wallet_grpc_test.rs#L102-L103